### ISSUE TITLE
Empty remote push url entry is ignored

### ISSUE THREAD

**Issue by rcorre** [libgit2/libgit2#2968]
On git CLI, you can explicitly set an empty push URL:

    git remote add foo some.url/here
    git remote set-url --push foo ""

This is interpreted by core git as a remote with an empty push url.

    > git remote -v
    foo some.url/here (fetch)
    foo   (push)

libgit2 (`master`) just uses the fetch url to set the push, as it [ignores the empty push entry](https://github.com/libgit2/libgit2/blob/f438dbe1910fcb39d8d883543663274f58405bca/src/remote.c#L432-L435).

This is a pretty weird edge case, but it is inconsistent with core git.

**Comment by carlosmn** [comment]
How does git react in this case? Is the behaviour with an empty path documented or tested in git itself?

**Comment by rcorre** [comment]
```
> git config -l | grep pushurl
remote.foo.pushurl=
> git push foo
fatal: The current branch master has no upstream branch.
```

This is different from the error message when the remote is non-empty but invalid (<url> does not appear to be a git repository).

While the manpage doesn't specifically say anything about empty urls, it does say this under the entry for `set-url`:
"Trying to delete all non-push URLs is an error".
However, deleting an entry with `git remote set-url --delete --push foo <url> doesn't set the url to empty, it sets the push url to match the fetch url.

**Comment by carlosmn** [comment]
That error message is about not having an upstream branch, which would depend on the push default and the configuration of the local branch. It looks like you have your `push.default` option to upstream, but your current branch does not have an upstream. This would not be related to the pushurl being empty.

Using `simple` or `matching` as push defaults still errors out, but with a topical error, as as when running `git push ''`.

    fatal: No path specified. See 'man git-pull' for valid url syntax

But I'm not convinced that this is something where the behaviour is defined. Unless  tested, it still looks to me like a

**Comment by rcorre** [comment]
I agree this falls pretty far in edge-case territory. My concern is that, as it stands, core git _does_ let you input an empty url, which generates a config entry that it _will_ read. If libgit2 ignores this entry, then you end up with a situation where git-core and libgit2 behave differently in the same repo.

**Comment by carlosmn** [comment]
We do read it, but without knowing exactly what's considered to be the fallback to default, it could be unwanted behaviour on either implementation.
